oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/core/expression/DefaultOAuth2Expressions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.expression;
```

Comment:
Why is this in `core` package?

Reply:
Right, good catch.

What do you think, are these expressions something that other OAuth2 actors (auth server, client) will want to be able to do?

Or are you thinking that if we support expressions for in those contexts, we would just create specific beans for each in their own libraries?

Reply:
sweet, I've moved this to `resourceserver.access.expression`